sys/net/network_layer/sixlowpan/lowpan.c

```diff
@@ -1110,7 +1110,7 @@ uint8_t lowpan_iphc_encoding(int if_id, const uint8_t *dest, int dest_len,
             else if (dest_len == 2 &&
                      ipv6_buf->destaddr.uint32[2] == HTONL(0x000000ff) &&
                      ipv6_buf->destaddr.uint16[6] == HTONS(0xfe00) &&
-                     ipv6_buf->destaddr.uint16[7] == *((uint16_t *) dest)) {
+                     ipv6_buf->destaddr.uint16[7] == (dest[1] << 8 | dest[0])) {
```

Comment:
You need to use the byte order macros when you assign dest.u16 in this way.
On Mar 28, 2015 1:29 PM, "Lucas Jenß" <notifications@github.com> wrote:

> In sys/net/network_layer/sixlowpan/lowpan.c
> <https://github.com/RIOT-OS/RIOT/pull/2727#discussion_r27344735>:
>
> > @@ -1110,7 +1110,7 @@ uint8_t lowpan_iphc_encoding(int if_id, const uint8_t *dest, int dest_len,
> >              else if (dest_len == 2 &&
> >                       ipv6_buf->destaddr.uint32[2] == HTONL(0x000000ff) &&
> >                       ipv6_buf->destaddr.uint16[6] == HTONS(0xfe00) &&
> > -                     ipv6_buf->destaddr.uint16[7] == *((uint16_t *) dest)) {
> > +                     ipv6_buf->destaddr.uint16[7] == (dest[1] << 8 | dest[0])) {
>
> Not sure if I agree. While it is more readable in the sense that less
> people are going to "stumble" over that line without knowing that exactly
> it does, I think that your version does not convey the actual intention of
> the comparison. Also, bitops are used quite a lot in this particular source
> file anyway.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RIOT-OS/RIOT/pull/2727/files#r27344735>.
>